lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -446,17 +447,21 @@ func selectHopHints(amtMSat lnwire.MilliSatoshi, cfg *AddInvoiceConfig,
 			continue
 		}
 
-		// Now that we now this channel use usable, add it as a hop
-		// hint and the indexes we'll use later.
-		addHopHint(&hopHints, channel, edgePolicy)
+		// Now that we now this channel use usable, add it to the
+		// policies map and the indexes we'll use later.
+		hopHintPolicies.addPolicy(channel, edgePolicy, amtMSat)
 
 		hopHintChans[channel.FundingOutpoint] = struct{}{}
 		totalHintBandwidth += channel.LocalCommitment.RemoteBalance
```

Comment:
note that this still adds the channel outpoint to hopHintChans and adds the remote capacity to the total bandwidth even though the 'addPolicy' method doesnt necessarily add this new channel as a hop hint. This should be fine since, due to non-strict forwarding, the total bandwidth is in essence the sum of all the channel capacities even if only 1 hop hint is given

Reply:
Hm, been thinking about this a bit, and looking at `htlcswitch/switch.go#handlePacketForward` (relevant code is the `UpdateAddHTLC` [case](https://github.com/lightningnetwork/lnd/blob/master/htlcswitch/switch.go#L961). In this context link basically means channel. 

When a lnd node (B) gets a payment which they need to forward along multiple private channels to the next hop (A), we try to use our target link, and the fallback to any link with the same peer which has sufficient bandwidth. So while we use maximum timeout/highest fees, we don't actually get to use the sum of all the channels we have with that peer. Does that make sense? The switch is one of the more complex parts of lnd to wrap your head around so happy to chat about it more. 

So where this could be wrong is if we have multiple channels with a peer, and then we exit the hop hint loop early (since we counted channel sum rather than largest channel). This all gets a bit subset-summey quickly (a curse of lightning dev). A solution here could be to return a "bandwidth delta" from `hopHintPolicies.addPolicy` so that we can replace the previous max chan balance with our new one (if bigger)? Need to think about it a bit more, could influence the way we approach this as well. 

The concept where we would have total bandwidth available is something called link-level-mpp, and would require forwarding nodes to split the htlc themselves, and the receiving node to re-assemble on their end. Which we don't have (yet) :(

Reply:
ok cool, yeah that makes sense. I will try the 'bandwidth delta' way you suggested 👍